tool/tgz/testing.go

```diff
@@ -0,0 +1,63 @@
+package tgz
+
+import (
+	"archive/tar"
+	"bytes"
+	"image"
+	"image/color"
+	"image/draw"
+	"image/png"
+	"path/filepath"
+	"testing"
+)
+
+// SynthesizeTarball generates a tar.gz file for testing.
+func SynthesizeTarball(t *testing.T, dir string) string {
+	fn := filepath.Join(dir, "input.tar.gz")
+
+	w, e := CreateFile(fn)
+	if e != nil {
+		t.Fatalf("Cannot create writer: %v", e)
+	}
+
+	imgFns := []string{
+		"mnist/training/0/first.png",
+		"mnist/training/0/second.png",
+		"mnist/training/1/first.png",
+		"mnist/training/1/second.png",
+		"mnist/testing/0/first.png",
+	}
```

Comment:
Maybe we can add some folders that would be more authentically.

``` golang
imgFolders := []string {
    "mnist",
     "mnist/training/",
     "mnist/testing",
     "mnist/training/0",
     "mnist/training/1",
      "mnist/testing/0"
}
```

Reply:
Done